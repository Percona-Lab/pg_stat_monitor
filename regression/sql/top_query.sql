CREATE EXTENSION pg_stat_monitor;
SELECT pg_stat_monitor_reset();

CREATE OR REPLACE FUNCTION add(int, int) RETURNS INTEGER AS
$$
BEGIN
	return (select $1 + $2);
END; $$ language plpgsql;

CREATE OR REPLACE function add2(int, int) RETURNS int as
$$
BEGIN
	return add($1,$2);
END;
$$ language plpgsql;

SELECT add2(1,2);
SELECT query, top_query FROM pg_stat_monitor ORDER BY query COLLATE "C";

ALTER SYSTEM SET pg_stat_monitor.track TO 'all';
SELECT pg_reload_conf();
SELECT pg_sleep(1);
SELECT pg_stat_monitor_reset();

CREATE OR REPLACE FUNCTION add(int, int) RETURNS INTEGER AS
$$
BEGIN
	return (select $1 + $2);
END; $$ language plpgsql;

CREATE OR REPLACE function add2(int, int) RETURNS int as
$$
BEGIN
	return add($1,$2);
END;
$$ language plpgsql;

SELECT add2(1,2);
SELECT query, top_query FROM pg_stat_monitor ORDER BY query COLLATE "C";
ALTER SYSTEM SET pg_stat_monitor.track TO 'top';
SELECT pg_reload_conf();
SELECT pg_sleep(1);
SELECT pg_stat_monitor_reset();

DROP EXTENSION pg_stat_monitor;
