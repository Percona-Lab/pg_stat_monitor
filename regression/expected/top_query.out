CREATE EXTENSION pg_stat_monitor;
SELECT pg_stat_monitor_reset();
 pg_stat_monitor_reset 
-----------------------
 
(1 row)

CREATE OR REPLACE FUNCTION add(int, int) RETURNS INTEGER AS
$$
BEGIN
	return (select $1 + $2);
END; $$ language plpgsql;
CREATE OR REPLACE function add2(int, int) RETURNS int as
$$
BEGIN
	return add($1,$2);
END;
$$ language plpgsql;
SELECT add2(1,2);
 add2 
------
    3
(1 row)

SELECT query, top_query FROM pg_stat_monitor ORDER BY query COLLATE "C";
                                  query                                  | top_query 
-------------------------------------------------------------------------+-----------
 CREATE OR REPLACE FUNCTION add(int, int) RETURNS INTEGER AS            +| 
 $$                                                                     +| 
 BEGIN                                                                  +| 
         return (select $1 + $2);                                       +| 
 END; $$ language plpgsql                                                | 
 CREATE OR REPLACE function add2(int, int) RETURNS int as               +| 
 $$                                                                     +| 
 BEGIN                                                                  +| 
         return add($1,$2);                                             +| 
 END;                                                                   +| 
 $$ language plpgsql                                                     | 
 SELECT add2($1,$2)                                                      | 
 SELECT pg_stat_monitor_reset()                                          | 
 SELECT query, top_query FROM pg_stat_monitor ORDER BY query COLLATE "C" | 
(5 rows)

ALTER SYSTEM SET pg_stat_monitor.track TO 'all';
SELECT pg_reload_conf();
 pg_reload_conf 
----------------
 t
(1 row)

SELECT pg_sleep(1);
 pg_sleep 
----------
 
(1 row)

SELECT pg_stat_monitor_reset();
 pg_stat_monitor_reset 
-----------------------
 
(1 row)

CREATE OR REPLACE FUNCTION add(int, int) RETURNS INTEGER AS
$$
BEGIN
	return (select $1 + $2);
END; $$ language plpgsql;
CREATE OR REPLACE function add2(int, int) RETURNS int as
$$
BEGIN
	return add($1,$2);
END;
$$ language plpgsql;
SELECT add2(1,2);
 add2 
------
    3
(1 row)

SELECT query, top_query FROM pg_stat_monitor ORDER BY query COLLATE "C";
                                  query                                  |     top_query      
-------------------------------------------------------------------------+--------------------
 CREATE OR REPLACE FUNCTION add(int, int) RETURNS INTEGER AS            +| 
 $$                                                                     +| 
 BEGIN                                                                  +| 
         return (select $1 + $2);                                       +| 
 END; $$ language plpgsql                                                | 
 CREATE OR REPLACE function add2(int, int) RETURNS int as               +| 
 $$                                                                     +| 
 BEGIN                                                                  +| 
         return add($1,$2);                                             +| 
 END;                                                                   +| 
 $$ language plpgsql                                                     | 
 SELECT (select $1 + $2)                                                 | SELECT add2($1,$2)
 SELECT add2($1,$2)                                                      | 
 SELECT pg_stat_monitor_reset()                                          | 
 SELECT query, top_query FROM pg_stat_monitor ORDER BY query COLLATE "C" | 
(6 rows)

ALTER SYSTEM SET pg_stat_monitor.track TO 'top';
SELECT pg_reload_conf();
 pg_reload_conf 
----------------
 t
(1 row)

SELECT pg_sleep(1);
 pg_sleep 
----------
 
(1 row)

SELECT pg_stat_monitor_reset();
 pg_stat_monitor_reset 
-----------------------
 
(1 row)

DROP EXTENSION pg_stat_monitor;
